src/main/java/com/hivemq/persistence/clientsession/ClientSession.java

```diff
@@ -17,30 +17,36 @@
 package com.hivemq.persistence.clientsession;
 
 import com.google.common.base.Preconditions;
+import com.hivemq.extension.sdk.api.annotations.NotNull;
+import com.hivemq.persistence.Sizable;
+import com.hivemq.util.ObjectMemoryEstimation;
 
 import static com.hivemq.mqtt.message.connect.Mqtt5CONNECT.SESSION_EXPIRE_ON_DISCONNECT;
 
 /**
  * @author Lukas Brandl
  */
-public class ClientSession {
+public class ClientSession implements Sizable {
 
     private boolean connected;
-
     private long sessionExpiryInterval;
 
+    private volatile int inMemorySize = SIZE_NOT_CALCULATED;
```

Comment:
volatile is unnecessary, the worst that can happen is that the size is calculated multiple times by different threads, but the result should be the same